apps/studio/components/interfaces/Integrations/CronJobs/CronJobScheduleSection.tsx

```diff
@@ -76,47 +78,22 @@ export const CronJobScheduleSection = ({ form, supportsSeconds }: CronJobSchedul
   })
 
   useEffect(() => {
-    if (!useNaturalLanguage || !debouncedValue) return
-    generateCronSyntax(debouncedValue)
+    if (useNaturalLanguage && debouncedValue) {
+      generateCronSyntax(debouncedValue)
+      return () => stop()
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [debouncedValue, useNaturalLanguage])
 
-  useEffect(() => {
-    if (!inputValue || inputValue.length < 5) return // set a min length before showing invalid message
-
-    // update the cronstrue string when the input value changes
-    try {
-      setScheduleString(CronToString(inputValue))
-      form.setValue('schedule', inputValue)
-    } catch (error) {
-      console.error('Error converting cron expression to string:', error)
-    }
-  }, [form, inputValue])
-
-  useEffect(() => {
-    if (useNaturalLanguage) return
-
-    setPresetValue(schedule)
-
-    if (!schedule) {
-      setScheduleString('')
-      return
-    }
-
-    try {
-      // Don't allow seconds-based schedules if seconds aren't supported
-      if (!supportsSeconds && secondsPattern.test(schedule)) {
-        setScheduleString('Invalid cron expression')
-        return
-      }
-
-      setScheduleString(CronToString(schedule))
-    } catch (error) {
-      setScheduleString('Invalid cron expression')
-      console.error('Error converting cron expression to string:', error)
+  const schedule = form.watch('schedule')
+  let scheduleString = ''
```

Comment:
hmm can we use ternary for this? we can wrap the `.test` in a separate function that has a try catch

in favor of immutability + consistency

Reply:
pushed a fix for this!